packages/api/src/hooks/p2p/useAdvertInfo.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import useQuery from '../../useQuery';
+
+/**
+ * This custom hook returns the advert information about the given advert ID.
+ */
+const useAdvertInfo = (
+    payload: NonNullable<Parameters<typeof useQuery<'p2p_advert_info'>>[1]>['payload'],
+    options?: NonNullable<Parameters<typeof useQuery<'p2p_advert_info'>>[1]>['options']
+) => {
+    const { data, ...rest } = useQuery('p2p_advert_info', {
+        payload: { id: payload?.id, use_client_limits: payload?.use_client_limits },
```

Comment:
```suggestion
        payload,
```

Base on our discussion 🤔